TACC/Core-CMS
# feat(djangocms_blog): multimedia article styles

## Overview

Style blog articles in a "Multimedia" category.

## Related

- [**TUP-118**](https://jira.tacc.utexas.edu/browse/TUP-118) ([TUP-258](https://jira.tacc.utexas.edu/browse/TUP-258))
- required by https://github.com/TACC/tup-ui/pull/98

## Changes

- changed `.app-blog` to be a container for absolute positioned of blog elements
- added styles for articles in a "Multimedia" / `multimedia` category
- added class names, for categories and tags, to the `.app-blog` element
- fixed class names, for categories and tags, on the category and tag links
- added URL to `post_detail.html` source template

## Testing

1. Add/Create local CMS.
2. [Install blog.](https://github.com/TACC/Core-CMS/wiki/How-to-Install-DjangoCMS-Blog-i.e.-News)*
3. Add an Article with:
    - Title†
    - Subtitle†
    - Abstract†
    - a "Published since" Date
    - a "Published since" Time
4. Add a Category named "Multimedia".
5. Assign that Article to that Category.
6. Load that article.
7. Add a large media in the Media placeholder.†
8. Add text content to the Article.†
9. Confirm visual looks similar to [design].

[design]: https://xd.adobe.com/view/6947ba13-16f1-4ff9-b1f0-0ed3ffcdc20b-3c14/screen/38333693-be4d-489c-a476-999d2032b5b4/specs/

<sup>* If you have trouble, see [(Confluence) Django CMS - Developer Guide - Blog/News](https://confluence.tacc.utexas.edu/pages/viewpage.action?pageId=265425776).
† I used content from https://www.tacc.utexas.edu/-/final-dance-of-unequal-black-hole-partners which is also (temporarily) available at https://dev.tup.tacc.utexas.edu/news/latest-news/2020/11/07/final-dance-unequal-black-hole-partners/.</sup>

## UI

| Before | After | [Design](https://xd.adobe.com/view/6947ba13-16f1-4ff9-b1f0-0ed3ffcdc20b-3c14/screen/38333693-be4d-489c-a476-999d2032b5b4/specs/)
| - | - | -
| ![before](https://user-images.githubusercontent.com/62723358/206266336-0dca48d1-c78c-4432-9904-a2e50a80968a.png) | ![after](https://user-images.githubusercontent.com/62723358/206269218-f539e4e5-4907-49f9-8015-b4be09b4282e.png) | ![design 1](https://user-images.githubusercontent.com/62723358/206266744-d60f10cc-e754-4892-aeea-466e4ab0ce6d.png) ![design 2](https://user-images.githubusercontent.com/62723358/206266918-b00e27c1-45e9-4fd4-8d8a-34e44ecce974.png)

| dev.tup test (via Snippet mirroring b7ca806) |
| - |
| ![tup dev blog multimedia article](https://user-images.githubusercontent.com/62723358/211915795-6e64c79e-1623-4aaf-a80b-fc593bce4161.png) |

## Notes

### Known Issues

1. Style is based off of [Texascale's design](https://xd.adobe.com/view/6947ba13-16f1-4ff9-b1f0-0ed3ffcdc20b-3c14/screen/38333693-be4d-489c-a476-999d2032b5b4/specs/). <sub>Style minutia (e.g. issue 2) can be performed when I have a TACC design.</sub>
2. No style for category links (i.e. "Multimedia") nor tag links (i.e. "multimedia").
